### ISSUE TITLE
Ability to handle ACL creation

### ISSUE THREAD

**Issue by pmonks** [pmonks/alfresco-bulk-import#8]
Migrated from https://code.google.com/p/alfresco-bulk-filesystem-import/issues/detail?id=100

It would be ideal if the bulk filesystem import tool could also handle ACLs.  Possibly add a configuration in the properties.xml file for each ACE.

**Comment by mayasd** [comment]
Hi,
It's not still possible to import ACLs with properties.xml ?

**Comment by pmonks** [comment]
That's correct. You should watch this ticket in order to be notified of any work here.

In the meantime you can post-process the imported content to set ACLs. This is a relatively common activity and one that Alfresco's SI partners and/or prof. svcs. team can assist with.

**Comment by zhihailiu** [comment]
Peter, is there a plan to tackle this issue anytime soon? In addition to the post-process approach as you suggested, if anyone wants to roll up his/her sleeves, does the bulk import tool code provide any API or hook to support it? You know, something similar to onCreateNode in Alfresco Behavior, or AOP interceptor? Appreciate your help.

**Comment by pmonks** [comment]
I don't have any immediate plan to fix it, for two reasons:
1. I consider post-processing to be a better approach, in part because...
2. Richard McKnight implemented this for v1.x of the tool and found that the performance impact was dire. IIRC it seems that Alfresco doesn't expect ACLs to change rapidly, and doing so causes issues with database and cache thrashing. 

That said, I'd be open to PRs that added this capability in a way that didn't impact import performance for those who don't need to set ACLs and/or are happy to set them via post-processing.
